holoviews/core/data/interface.py

```diff
@@ -307,25 +297,35 @@ def range(cls, dataset, dimension):
                 return column[0], column[-1]
 
     @classmethod
-    def concatenate(cls, dataset, datatype=None):
+    def concatenate(cls, datasets, datatype=None, new_type=None):
         """
-        Utility function to concatenate a list of Column objects,
-        returning a new Dataset object. Note that this is unlike the
-        .concat method which only concatenates the data.
+        Utility function to concatenate an NdMapping of Dataset objects.
         """
-        if len(set(type(c) for c in dataset)) != 1:
-               raise Exception("All inputs must be same type in order to concatenate")
-
-        interfaces = set(c.interface for c in dataset)
-        if len(interfaces)!=1 and datatype is None:
-            raise Exception("Please specify the concatenated datatype")
-        elif len(interfaces)!=1:
-            interface = cls.interfaces[datatype]
-        else:
-            interface = interfaces.pop()
-
-        concat_data = interface.concat(dataset)
-        return dataset[0].clone(concat_data)
+        from . import Dataset, default_datatype
+        new_type = new_type or Dataset
+        if isinstance(datasets, NdMapping):
+            dimensions = datasets.kdims
+            datasets = datasets.data
+        if isinstance(datasets, (dict, OrderedDict)):
+            datasets = datasets.items()
+            keys, datasets = zip(*datasets)
+        elif isinstance(datasets, list) and not any(isinstance(v, tuple) for v in datasets):
+            keys = [()]*len(datasets)
```

Comment:
What are all these empty tuple keys for? Just to get things in the right format?

Reply:
Right, concatenate is usually meant for concatenating along some dimension but you can also concatenate a simple list of datasets without concatenating along some dimensions. For that case we generate empty tuple keys. Happy to add a comment. Separately I also need to assert that this only happens for tabular data, since gridded data must be concatenated along some dimension.